feature/experimental/system/health/tests/system_generic_health_check/system_generic_health_check_test.go

```diff
@@ -0,0 +1,753 @@
+	}
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	cpuCards := components.FindComponentsByType(t, dut, cpuType)
+	for _, cpu := range cpuCards {
+		query := gnmi.OC().Component(cpu).State()
+		timestamp := time.Now().Round(time.Second)
+		component := gnmi.Get(t, dut, query)
+		cpuParent := component.GetParent()
+		if cpuParent == "" {
+			t.Errorf("ERROR: can't find parent information for CPU card %v", component)
+		}
+
+		if slices.Contains(controllerCards, cpuParent) {
+			// Remove parent from the list of check cards.
+			controllerCards = removeElement(controllerCards, cpuParent)
+			cpuUtilization := component.GetCpu().GetUtilization()
+			if cpuUtilization.Avg == nil {
+				t.Errorf("ERROR: %s %s %s Type %-20s - CPU utilization data not available",
+					timestamp, deviceName, cpu, cpuParent)
+				continue
+			}
+			t.Logf("INFO: %s %s Type %-20s %-10s - Utilization: %3d%%", timestamp, deviceName, cpu, cpuParent, cpuUtilization.GetAvg())
+		}
+	}
+	if len(controllerCards) > 0 {
+		t.Errorf("ERROR: Didn't find cpu card for checkCards %s", controllerCards)
+	}
+}
+
+func TestLineCardsNoHighCPUSpike(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+
+	if deviations.LinecardCPUUtilizationUnsupported(dut) {
+		t.Skipf("Skipping test due to deviation linecard_cpu_ultilization_unsupported")
+	}
+
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	cpuCards := components.FindComponentsByType(t, dut, cpuType)
+	for _, cpu := range cpuCards {
+		timestamp := time.Now().Round(time.Second)
+
+		query := gnmi.OC().Component(cpu).State()
+		component := gnmi.Get(t, dut, query)
+
+		cpuParent := component.GetParent()
+		if cpuParent == "" {
+			t.Errorf("ERROR: can't find parent information for CPU card %v", component)
+		}
+
+		// If cpu card's parent is line card, check cpu ultilization.
+		if slices.Contains(lineCards, cpuParent) {
+			// Remove parent from the list of check cards.
+			lineCards = removeElement(lineCards, cpuParent)
+			// Fetch CPU utilization data.
+			cpuUtilization := component.GetCpu().GetUtilization()
+			if cpuUtilization.Avg == nil {
+				t.Errorf("ERROR: %s %s %s Type %-20s - CPU utilization data not available",
+					timestamp, deviceName, cpu, cpuParent)
+				continue
+			}
+			t.Logf("INFO: %s %s Type %-20s %-10s - Utilization: %3d%%", timestamp, deviceName, cpu, cpuParent, cpuUtilization.GetAvg())
+		}
+	}
+
+	if len(lineCards) > 0 {
+		t.Errorf("ERROR: Didn't find cpu card for checkCards %s", lineCards)
+	}
+}
+
+func TestComponentsNoHighMemoryUtilization(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	deviceName := dut.Name()
+	description := "Component"
+
+	controllerCards := components.FindComponentsByType(t, dut, controllerCardType)
+	lineCards := components.FindComponentsByType(t, dut, lineCardType)
+	cardList := append(controllerCards, lineCards...)
+	for _, component := range cardList {
```

Comment:
t.Run()

Case for no cards scenario.

Reply:
Thanks for your comments. Add t.Run() and no cards scenario checking:

```go
	cardList := append(controllerCards, lineCards...)
	if len(cardList) == 0 {
		t.Errorf("ERROR: No card has been found.")
	}
	for _, component := range cardList {
		t.Run(component, func(t *testing.T) {
```